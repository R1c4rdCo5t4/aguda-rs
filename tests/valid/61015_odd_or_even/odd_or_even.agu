-- Author: 61015, Filipe Lopes

let evenOddDesc (n : Int) : Int -> String =
  if n % 2 == 0 then "Even" else "Odd";

let i : Int = 1;
let evenSum : Int = 0;
let oddSum : Int = 0;
let evenCount : Int = 0;
let oddCount : Int = 0;

while i <= 10 do (
  print("Num: "); print(i); print(" "); print(evenOddDesc(i)); print("\n");
  if i % 2 == 0 then (
    set evenSum = evenSum + i;
    set evenCount = evenCount + 1
  ) else (
    set oddSum = oddSum + i;
    set oddCount = oddCount + 1
  );
  set i = i + 1
);

if evenCount > 0 then (
  let evenAvg : Int = evenSum / evenCount;
  print("Even avg: "); print(evenAvg); print("\n")
);

if oddCount > 0 then (
  let oddAvg : Int = oddSum / oddCount;
  print("Odd avg: "); print(oddAvg); print("\n")
);

print("Even sum: "); print(evenSum); print(" Odd sum: "); print(oddSum); print("\n");