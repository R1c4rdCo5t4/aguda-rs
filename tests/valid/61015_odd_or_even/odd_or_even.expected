Expected output: Num: 1 Odd
                 Num: 2 Even
                 Num: 3 Odd
                 Num: 4 Even
                 Num: 5 Odd
                 Num: 6 Even
                 Num: 7 Odd
                 Num: 8 Even
                 Num: 9 Odd
                 Num: 10 Even
                 Even avg: 6
                 Odd avg: 5
                 Even sum: 30 
                 Odd sum: 25